mitosheet/src/mito/Mito.tsx

```diff
@@ -334,54 +332,6 @@ export const Mito = (props: MitoProps): JSX.Element => {
         previousNumSheetsRef.current = sheetDataArray.length;
     }, [sheetDataArray])
 
-    const previousNumGraphsRef = useRef<number>(Object.keys(analysisData.graphDataDict || {}).length)
-    const previousGraphIndex = useRef<number>(uiState.selectedGraphID !== undefined ?
-        Object.keys(analysisData.graphDataDict  || {}).indexOf(uiState.selectedGraphID) : -1)
-
-    // When we switch graphID's make sure that we keep the previousGraphIndex up to date
-    useEffect(() => {
-        previousGraphIndex.current = uiState.selectedGraphID !== undefined ?
-            Object.keys(analysisData.graphDataDict || {}).indexOf(uiState.selectedGraphID) : -1
-    }, [uiState.selectedGraphID])
-
-    // Update the selected sheet tab when the number of graphs change. 
-    useEffect(() => {
-        const graphIDs = Object.keys(analysisData.graphDataDict || {})
-        const previousNumGraphs = previousNumGraphsRef.current;
-        const newNumGraphs = Object.keys(analysisData.graphDataDict || {}).length
-
-        // Handle new graph created
-        if (previousNumGraphs < newNumGraphs) {
-            const newGraphID = graphIDs[newNumGraphs - 1]
-            setUIState(prevUIState => {
-                return {
-                    ...prevUIState,
-                    selectedGraphID: newGraphID,
-                    selectedTabType: 'graph',
-                    currOpenTaskpane: {
-                        type: TaskpaneType.GRAPH,
-                        graphID: newGraphID,
-                    },
-                }
-            })
-
-            // Update the previous graph index for next time
-            previousGraphIndex.current = graphIDs.indexOf(newGraphID)
-
-        // Handle graph removal
-        } else if (previousNumGraphs > newNumGraphs) {
```

Comment:
Also, I wonder about this case -- what if there are graphs removed -- do we need to close the sidebar?

Reply:
Honestly, both of these might be better if they were not in the effect -- but wondering if we handle them.